pkg/identity/github/issuer.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/sigstore/fulcio/pkg/identity"
+)
+
+type actionsIssuer struct {
+	*oidc.IDTokenVerifier
+}
+
+func NewActionsIssuer(clientID string) (identity.Issuer, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	provider, err := oidc.NewProvider(ctx, issuerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return &actionsIssuer{
+		IDTokenVerifier: provider.Verifier(&oidc.Config{ClientID: clientID}),
+	}, nil
+}
+
+func (i *actionsIssuer) Match(_ context.Context, url string) bool {
+	return url == issuerURL
+}
+
+func (i *actionsIssuer) Authenticate(ctx context.Context, token string) (identity.Principal, error) {
+	// Authenticate token first by checking ID token signature
+	parsed, err := i.Verify(ctx, token)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse claims on authenticated token
+	var claims struct {
+		JobWorkflowRef string `json:"job_workflow_ref"`
```

Comment:
nit: I think the rest of the code base uses `'`, any reason to switch to `?

Reply:
All the struct tags us ` I believe so you don't need to escape the ". I'm guessing you mean my ` usage for strings instead of " elsewhere? Its just a habit of mine to use them if I'm not using escape characters but I can change that to be in keeping with the rest of the code base